Automattic/woocommerce-payments
# Multicurrency: Do not modify deposit percentage for display.

Fixes: WooCommerce Deposits #\506
Supersedes PR #6930

#### Changes proposed in this Pull Request







As reported in  WooCommerce Deposits #\506, enabling multi-currency modifies the display of percentage deposits.

The issue comes from `WooCommerceDeposits::modify_cart_item_deposit_amount_meta()` attempting to convert a percentage number although it were currency.

A store with a USD base selling a product with a 20% deposit results in the following displaying for select currencies:

AUD: 31%
CAD: 27%
INR: 1656%

I haven't written any tests as there are existing tests for multicurrency that are passing but as the plugin is using mocks the tests are passing as they're failing to account for the various filters running in this and deposits.

This is a sibling PR for https://github.com/woocommerce/woocommerce-deposits/pull/544





**Product page**
<img width="923" alt="Screen Shot 2023-09-21 at 2 18 54 pm" src="https://github.com/Automattic/woocommerce-payments/assets/519727/9d6fdb57-dade-4bc9-9de7-a80d489a6385">

**Cart page**

<img width="485" alt="Screen Shot 2023-09-21 at 2 19 31 pm" src="https://github.com/Automattic/woocommerce-payments/assets/519727/42c222e1-3cd9-4531-8f8f-931526ba3733">

**Checkout page**

<img width="914" alt="Screen Shot 2023-09-21 at 2 20 21 pm" src="https://github.com/Automattic/woocommerce-payments/assets/519727/f3611c4b-a4f8-487b-8c5b-b5de1162ca01">

**Account > Order page**

<img width="705" alt="Screen Shot 2023-09-21 at 2 20 58 pm" src="https://github.com/Automattic/woocommerce-payments/assets/519727/c0b31488-bdbf-4989-a9c3-af9219b578ae">


#### Testing instructions



1. Configure Multi-currency Payments: using USD (default) and INR will best demonstrate the bug
2. Check out the Deposits extension branch in https://github.com/woocommerce/woocommerce-deposits/pull/544
3. Build the Deposits extension following the instructions in the readme.
4. Configure and publish product to accept deposits:
    * Enable Deposits: Yes - Deposits are optional
    * Deposit Type:: percentage
    * Deposit Amount: 20
    * Default Deposit Selected Type: Pay Deposit
5. View the product on the front end
6. Switch to USD if required, observe the deposit amount is displayed as 20%
7. Switch the currency from USD to INR
8. On this branch the deposit amount should remain 20%, on trunk the amount is significantly over 20%



*

-------------------


**Post merge**
